tree/ntuple/v7/inc/ROOT/RField/RFieldSequenceContainer.hxx

```diff
@@ -287,6 +291,21 @@ public:
    }
 };
 
+/// The generic field for a (nested) std::vector<Type> except for std::vector<bool>
```

Comment:
What is the purpose of this class?  It seems that it could even be simply:
```
using RVectorField = RSequenceCollectionField;
```

Reply:
The `RVectorField` class sets a type name (`std::vector<$innertype>`). The RSequenceCollectionField sets an empty type name.

Reply:
Can you point me where the type name is set to the different values? (Sorry if it is obvious, I am not seeing it today).

Reply:
Right now it is here: https://github.com/root-project/root/pull/16478/files#diff-1a2cfcc415e3af5e10c390121775f2e2865ae357dacee02772a4872cd7620e52R2490

Reply:
That's another option, if we make sure that users cannot set an arbitrary type name. Maybe with a `static` method `RVectorField::CreateUntyped(std::string_view fieldName, std::unique_ptr<RFieldBase> itemField)`? I also liked this functionality in a separate class, but maybe it's indeed not needed after all...

Reply:
Is `Unnamed/Untyped` the actual semantic or would 'Anonymous' (with the C++ semantic) be more flexible and/or accurate.  (For example, if this is for "leaflist" type of thing, one could imagine that at a later time one would want to read a file written with an untyped column into a real object with the equivalent schema).

Looking at the current code, it feels weird that the only 2 type name options for `RSequenceCollectionField` are empty and `std::vector<T>` (whereas the name sounds more generic).

What does the untyped/anonymous case return for `itemField->GetTypeName()`, i.e. couldn't we have simply:

```
ROOT::Experimental::RSequenceCollectionField::RSequenceCollectionField(std::string_view fieldName,
                                                                       std::unique_ptr<RFieldBase> itemField)
   : ROOT::Experimental::RFieldBase(fieldName, itemField->GetTypeName().empty() ? "" : "std::vector<" + itemField->GetTypeName() + ">",
```

Reply:
I think `anonymous` is wrong because the fields have a _field name_, just no _type name_. The `RSequenceCollectionField` is now removed and the "untyped functionality" is squashed into the `RVectorField`.